apm-agent-core/src/test/java/co/elastic/apm/agent/testutils/assertions/metrics/MetricsetsAssert.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.testutils.assertions.metrics;
+
+import co.elastic.apm.agent.testutils.assertions.BaseAssert;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+public class MetricsetsAssert extends BaseAssert<MetricsetsAssert, List<MetricsetJson>> {
```

Comment:
[minor] rename to `MetricSetAssert` for consistency: others similar assertions are in singular form + case.

Reply:
I fixed the casing. Regarding plural/singular: We have both `MetricSetAssert` and `MetricSetsAssert`: The first one has verifications for a single metricset, the second one for collections of metricsets.